include/envoy/http/header_map.h

```diff
@@ -129,6 +129,14 @@ class HeaderString {
    */
   Type type() const { return type_; }
 
+  /**
+   * Move the rhs value into this HeaderString.
+   * @param rhs an existing HeaderString whose contents MUST point to data that will live beyond
+   *        the lifetime of any request/response using the string (since a codec may optimize for
+   *        zero copy).
+   */
+  HeaderString& operator=(HeaderString&& rhs);
```

Comment:
Do we explicitly need the assignment operator? Would prefer not having compiler surprises here. Seems like we can basically just use the move function below? (Also, not sure comment about lifetime is accurate, the impl seems to handle all cases).